eureka2-write-server/src/main/java/com/netflix/eureka2/server/channel/RegistrationChannelImpl.java

```diff
@@ -99,10 +100,10 @@ public void onNext(Void aVoid) {
             }
 
             private void evictIfPresent() {
-                InstanceInfo toEvict = instanceInfoRef.get();
+                InstanceInfo toEvict = registrationSubject.getValue();
```

Comment:
I have missed the `@Experimental` annotation. Reading last value from subject let eliminate InstanceInfo field in the channel. I will have to put it back.